gen/generator.go

```diff
@@ -323,8 +323,8 @@ func LookupTestValue(classPkgName, propertyName string, testVars map[string]inte
 			}
 		}
 	}
-	_, ok = testVars["resource_required"]
-	if ok {
+
+	if _, ok := testVars["resource_required"]; ok && testVars["resource_required"] != nil {
```

Comment:
_, ok := testVars["resource_required"]; ----> will this not fail if -> testVars["resource_required"] = nil

Reply:
```
testVars := map[string]interface{}{
  "resource_required": "yes",
}

if _, ok := testVars["resource_required"]; ok && testVars["resource_required"] != nil {
  fmt.Println("----- inside if condition with valid value ----------")
}

testVars1 := map[string]interface{}{}

if _, ok := testVars1["resource_required"]; ok && testVars1["resource_required"] != nil {
  fmt.Println("----- inside if condition with invalid value ----------")
}


Output:
----- inside if condition with valid value ----------

```

Yes, the if condition checks will fail, and the code will not execute when the `testVars["resource_required"] = nil`.